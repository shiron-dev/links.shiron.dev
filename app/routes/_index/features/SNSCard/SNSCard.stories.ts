import type { Meta, StoryObj } from "@storybook/react";

import { createRemixStub } from "@remix-run/testing";
import { SNSCard } from "./SNSCard";

type T = typeof SNSCard;

const meta: Meta = {
  title: "SNS Card",
  component: SNSCard,
  decorators: [
    (story) => {
      const remixStub = createRemixStub([
        {
          Component: () => story(),
          path: "/*",
        },
      ]);

      return remixStub({});
    },
  ],
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {
  args: {
    name: "GitHub",
    userName: "@shiron-dev",
    icon: "/github-mark.svg",
    href: "https://github.com/shiron-dev",
  },
};

export default meta;
