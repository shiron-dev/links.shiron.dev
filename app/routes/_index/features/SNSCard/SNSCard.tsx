import { Link } from "@remix-run/react";
import * as styles from "./styles.css";

interface Props {
  name: string
  userName: string
  icon: string
  href: string
  width?: number
  height?: number
}

export function SNSCard(props: Props): JSX.Element {
  return (
    <Link to={props.href} className={styles.snsCard}>
      <img
        src={props.icon}
        alt={props.name}
        width={props.width ?? 64}
        height={props.height ?? 64}
      />
      <div>
        <div className="title">
          <div className="sns-name">{props.name}</div>
          <div className="user-name">{props.userName}</div>
        </div>
      </div>
    </Link>
  );
}
