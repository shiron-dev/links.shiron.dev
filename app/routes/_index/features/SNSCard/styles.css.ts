import { style } from "@vanilla-extract/css";

export const snsCard = style({
  display: "flex",
  // ":link": {
  //   color: "black",
  //   textDecoration: "none",
  // },
  // ":visited": {
  //   color: "black",
  //   textDecoration: "none",
  // },
  // ":hover": {
  //   color: "black",
  //   textDecoration: "none",
  // },
  // ":active": {
  //   color: "black",
  //   textDecoration: "none",
  // },
  // "selectors": {
  //   "& > *": {
  //     margin: "0 0.5rem",
  //   },
  //   "& > div": {
  //     alignSelf: "center",
  //   },
  // },
});

export const title = style({});

export const snsName = style({
  fontSize: "1.5rem",
});

export const userName = style({
  fontSize: "1rem",
  marginLeft: "0.5rem",
});
