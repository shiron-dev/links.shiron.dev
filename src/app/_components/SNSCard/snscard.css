.sns-card {
  display: flex;
}

.sns-card:link,
.sns-card:visited,
.sns-card:hover,
.sns-card:active {
  color: black;
  text-decoration: none;
}

.sns-card > * {
  margin: 0 0.5rem;
}

.sns-card > div {
  align-self: center;
}

.title {
}

.sns-name {
  font-size: 1.5rem;
}

.user-name {
  font-size: 1rem;
  margin-left: 0.5rem;
}
