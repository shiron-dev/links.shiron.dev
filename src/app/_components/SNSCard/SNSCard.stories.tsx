import type { Meta, StoryObj } from "@storybook/react";
import { SNSCard } from "./SNSCard";

const meta: Meta<typeof SNSCard> = {
  title: "components/SNSCard",
  component: SNSCard,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  args: {
    icon: "https://github.githubassets.com/favicons/favicon.svg",
    name: "GitHub",
    href: "/",
    userName: "username",
  },
};

export default meta;
type Story = StoryObj<typeof SNSCard>;

export const Default: Story = {};
