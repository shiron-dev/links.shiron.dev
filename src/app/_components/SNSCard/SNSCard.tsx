import Link from "next/link";
import Image from "next/image";
import "./snscard.css";

interface Props {
  name: string;
  userName: string;
  icon: string;
  href: string;
  width?: number;
  height?: number;
}

export const SNSCard = (props: Props): JSX.Element => {
  return (
    <Link href={props.href} className="sns-card">
      <Image
        src={props.icon}
        alt={props.name}
        width={props.width ?? 64}
        height={props.height ?? 64}
      />
      <div>
        <div className="title">
          <div className="sns-name">{props.name}</div>
          <div className="user-name">{props.userName}</div>
        </div>
      </div>
    </Link>
  );
};
