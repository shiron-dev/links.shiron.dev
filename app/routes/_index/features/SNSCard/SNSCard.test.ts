// @vitest-environment jsdom

import { composeStories } from "@storybook/react";

import { expect } from "vitest";

import * as stories from "./SNSCard.stories";

const { Default } = composeStories(stories);
it("button snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
