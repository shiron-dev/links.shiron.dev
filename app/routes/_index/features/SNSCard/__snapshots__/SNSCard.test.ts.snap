// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button snapshot 1`] = `
<div>
  <a
    class="styles_snsCard__mdbbxh0"
    href="https://github.com/shiron-dev"
  >
    <img
      alt="GitHub"
      height="64"
      src="/github-mark.svg"
      width="64"
    />
    <div>
      <div
        class="title"
      >
        <div
          class="sns-name"
        >
          GitHub
        </div>
        <div
          class="user-name"
        >
          @shiron-dev
        </div>
      </div>
    </div>
  </a>
</div>
`;
